File: f1-api/controllers/mock/classification.go
package mock

import "f1-api/model"

// ClassificationController :
type ClassificationController struct{}

// GetDriversClassification :
func (cc ClassificationController) GetDriversClassification() ([]model.Classification, error) {

	classification := []model.Classification{
		model.Classification{
			Name:   "Driver1",
			Points: 10,
		},
		model.Classification{
			Name:   "Driver2",
			Points: 5,
		},
	}

	return classification, nil
}

// GetTeamsClassification :
func (cc ClassificationController) GetTeamsClassification() ([]model.Classification, error) {

	classification := []model.Classification{
		model.Classification{
			Name:   "Team1",
			Points: 10,
		},
		model.Classification{
			Name:   "Team2",
			Points: 5,
		},
	}

	return classification, nil

}

File: f1-discord-bot/api-manager/classification.js
const request = require('./request.js')
const apiResources = require('./api-resources.js')
const classificationPosition = require('./../model/classificationPosition.js')


function getDriversClassification(serverOptions){

    console.log(`Requesting drivers classification from https://${serverOptions['hostname']}:${serverOptions['port']}${apiResources.driversClassification}`)

    let requestOptions = {
        hostname: serverOptions['hostname'],
        port: serverOptions['port'],
        path: apiResources.driversClassification,
        agent: false,
        method: 'GET',
        headers: {
            'Accept': 'application/json'
        }
    }

    return new Promise((resolve, reject) => {

        request.makeAsyncRequest(requestOptions).then(
            response => {

                console.log(`Success requesting drivers classificaction. Response: ${response}`)

                let driversClassification = []
                response['DriversClassification'].forEach(position => {
                    driversClassification.push(classificationPosition.ClassificationPosition.fromJSON(position))
                })

                resolve(driversClassification)
            }
        ).catch(
            error => {

                console.log(`Error requesting drivers classificaction. Error: ${error}`)
                reject(error)
            }
        )
    })
}


function getTeamsClassification(serverOptions){

    console.log(`Requesting teams classification from https://${serverOptions['hostname']}:${serverOptions['port']}${apiResources.teamsClassification}`)

    let requestOptions = {
        hostname: serverOptions['hostname'],
        port: serverOptions['port'],
        path: apiResources.teamsClassification,
        agent: false,
        method: 'GET',
        headers: {
            'Accept': 'application/json'
        }
    }

    return new Promise((resolve, reject) => {

        request.makeAsyncRequest(requestOptions).then(
            response => {

                console.log(`Success requesting teams classificaction. Response: ${response}`)

                let driversClassification = []
                response['TeamsClassification'].forEach(position => {
                    driversClassification.push(classificationPosition.ClassificationPosition.fromJSON(position))
                })
                resolve(driversClassification)
            }
        ).catch(
            error => {

                console.log(`Error requesting teams classificaction. Error: ${response}`)
                reject(error)
            }
        )
    }) 
}

module.exports = {
    getDriversClassification : getDriversClassification,
    getTeamsClassification: getTeamsClassification
}
